backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/DedicatedCpuPinningHelper.java

```diff
@@ -0,0 +1,152 @@
+package org.ovirt.engine.core.bll.scheduling.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.ovirt.engine.core.common.businessentities.CpuPinningPolicy;
+import org.ovirt.engine.core.common.businessentities.VDS;
+import org.ovirt.engine.core.common.businessentities.VM;
+import org.ovirt.engine.core.common.businessentities.VdsCpuUnit;
+import org.ovirt.engine.core.compat.Guid;
+
+public class DedicatedCpuPinningHelper {
+
+    public static synchronized boolean isDedicatedCpuPinningPossibleAtHost(List<VdsCpuUnit> cpuTopology,
```

Comment:
Could you please add a comment describing in detail the algorithm used to check if the dedicated cpu pinning is possible?